tools/eslint-rules/non-ascii-character.js

```diff
@@ -46,20 +46,10 @@ module.exports = (context) => {
       node,
       message,
       loc: sourceCode.getLocFromIndex(offendingCharacterPosition),
-      fix: (fixer) => {
-        return fixer.replaceText(
-          node,
-          suggestion ? `${suggestion}` : ''
-        );
-      }
     });
```

Comment:
We could let the fixer when it works, but removing it also works.
```suggestion
      fix: suggestion ?
        (fixer) => fixer.replaceText(node, suggestion) :
        undefined
    });
```

Reply:
That doesn't replace the single offending character. It replaces the text of the entire node. If you have `console.log("»");` in your code, it will replace your entire file with `"`.

Reply:
If we go through the trouble of fixing it so that it just replaces the offending character, it will still be broken because it will change `console.log("»");` to `console.log(""");` which is a syntax error. We'd basically only want to apply the suggestions in comments. Given how infrequently this rule comes up, I don't think spending time perfecting auto-fix on it is a good use of time (but I won't stop anyone else who wants to do it--I'm just unlikely to do it myself).